src/test/java/teammates/test/cases/webapi/RestoreCourseActionTest.java

```diff
@@ -0,0 +1,101 @@
+package teammates.test.cases.webapi;
+
+import org.apache.http.HttpStatus;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.exception.NullHttpParameterException;
+import teammates.common.util.Const;
+import teammates.logic.core.CoursesLogic;
+import teammates.storage.api.CoursesDb;
+import teammates.ui.webapi.action.JsonResult;
+import teammates.ui.webapi.action.RestoreCourseAction;
+import teammates.ui.webapi.output.MessageOutput;
+
+/**
+ * SUT: {@link RestoreCourseAction}.
+ */
+public class RestoreCourseActionTest
+        extends BaseActionTest<RestoreCourseAction> {
+
+    @Override
+    protected String getActionUri() {
+        return Const.ResourceURIs.BIN_COURSE;
+    }
+
+    @Override
+    protected String getRequestMethod() {
+        return DELETE;
+    }
+
+    @Override
+    @Test
+    public void testExecute() throws Exception {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        String instructorId = instructor1OfCourse1.googleId;
+        String courseId = instructor1OfCourse1.courseId;
+
+        loginAsInstructor(instructorId);
+
+        ______TS("Not in recycle bin but valid course");
+
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, courseId,
+        };
+
+        RestoreCourseAction action = getAction(submissionParams);
+        JsonResult result = getJsonResult(action);
+        MessageOutput message = (MessageOutput) result.getOutput();
+
+        assertEquals(HttpStatus.SC_OK, result.getStatusCode());
+        assertEquals("The course " + courseId + " has been restored.", message.getMessage());
```

Comment:
You should fetch the course and validate also? Same for other places?

Reply:
Will adding this `assertNull(CoursesLogic.inst().getSoftDeletedCourseForInstructor(instructor1OfCourse1));` be sufficient?

Reply:
1. You can convert `CoursesLogic.inst().` -> `logic.` for most methods
2. Try `AssertNull(logic.getCourse(instructor1OfCourse1.courseId).deletedAt);`

Reply:
yeah thats works! Thanks @ChooJeremy, have updated the tests accordingly